cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_predict_dense_default_batch_impl.i

```diff
@@ -694,7 +694,7 @@ Status PredictClassificationTask<algorithmFPType, cpu>::predictOneRowByAllTrees(
     return safeStat.detach();
 }
 
-#if defined(__INTEL_COMPILER)
```

Comment:
The check for `__INTEL_COMPILER` is sometimes replaced with `__AVX512F && DAAL_INTEL_CPP_COMPILER` and sometimes only with `DAAL_INTEL_CPP_COMPILER`. Why?

Reply:
as far i see there is behavior change between compilers and ICX is trying to compiler avx512 code even if it wasn't targeted for this code.  So AVX512 defines are put in place where we have direct AVX512 usage